client/src/main/java/io/getstream/chat/android/client/uploader/FileUploader.kt

```diff
@@ -0,0 +1,30 @@
+package io.getstream.chat.android.client.uploader
+
+import io.getstream.chat.android.client.call.Call
+import io.getstream.chat.android.client.utils.ProgressCallback
+import java.io.File
+
+public interface FileUploader {
```

Comment:
Shouldn't we add this when we already have a use for it? Kind of hard to review on its own.

Reply:
Jc asked to do this change (Improve CDN upload process) step-by-step and make PRs as small as possible.
Basically, `FileUploader` is going to wrap our `RetrofitCdnApi` and on the other hand - it will allow providing custom implementations